File: Exemplos/App_IntentDigitalHUB_Kotlin/Elgin_IntentDigitalHUB_Kotlin-font/app/src/main/java/com/elgin/intent_digital_hub/IntentDigitalHubService/BRIDGE/IniciaCancelamentoVenda.kt
package com.elgin.intent_digital_hub.IntentDigitalHubService.BRIDGE

import com.elgin.intent_digital_hub.IntentDigitalHubService.BRIDGE.BridgeCommand

class IniciaCancelamentoVenda(
    private val idTransacao: Int,
    private val pdv: String,
    private val valorTotal: String,
    private val dataHora: String,
    private val nsu: String
) : BridgeCommand("IniciaCancelamentoVenda") {
    override fun functionParameters(): String {
        return "\"idTransacao\"" + ":" + idTransacao + "," +
                "\"pdv\"" + ":" + "\"" + pdv + "\"" + "," +
                "\"valorTotal\"" + ":" + "\"" + valorTotal + "\"" + "," +
                "\"dataHora\"" + ":" + "\"" + dataHora + "\"" + "," +
                "\"nsu\"" + ":" + "\"" + nsu + "\""
    }
}

File: Exemplos M10/App_IntentDigitalHUB_Flutter/Elgin_IntentDigitalHUB_Flutter-font/lib/IntentDigitalHubService/BRIDGE/inicia_cancelamento_venda.dart
import 'package:intent_digital_hub/IntentDigitalHubService/BRIDGE/bridge_command.dart';

class IniciaCancelamentoVenda extends BridgeCommand {
  final int _idTransacao;
  final String _pdv;
  final String _valorTotal;
  final String _dataHora;
  final String _nsu;

  IniciaCancelamentoVenda(
      this._idTransacao, this._pdv, this._valorTotal, this._dataHora, this._nsu)
      : super('IniciaCancelamentoVenda');

  @override
  functionParameters() {
    return '"idTransacao"'
        ':'
        '$_idTransacao'
        ','
        '"pdv"'
        ':'
        '"$_pdv"'
        ','
        '"valorTotal"'
        ':'
        '"$_valorTotal"'
        ','
        '"dataHora"'
        ':'
        '"$_dataHora"'
        ','
        '"nsu"'
        ':'
        '"$_nsu"';
  }
}
